index.js

```diff
@@ -1,125 +1,201 @@
 'use strict'
 
 const nodePath = require('path')
-
-const commonExtname = require('common-extname')
-const commonPathPrefix = require('common-path-prefix')
-const glob = require('glob')
+const GlobSync = require('glob/sync').GlobSync
+const GLOBSTAR = require('minimatch').GLOBSTAR
+const hasMagic = require('glob').hasMagic
 const identifierfy = require('identifierfy')
-const flatten = require('lodash.flatten')
 
-function getPattern (path, source) {
-  const pattern = source.value.replace(/^glob:/, '').trim()
-  if (!pattern) {
-    throw path.buildCodeFrameError(`Missing glob pattern '${source.value}'`)
+const dirname = nodePath.dirname
+const relative = nodePath.relative
+const resolve = nodePath.resolve
+const fileSeparator = nodePath.sep
+
+function makeRe (set) {
+  let min, max
+  const parts = []
+  for (let i = 0; i < set.length; i++) {
+    const p = set[i]
+    if (typeof p !== 'string') {
+      if (p === GLOBSTAR) {
```

Comment:
Both `GLOBSTAR` and `typeof p !== 'string'` are regular expression - GLOBSTAR is just pre-defined. I want to capture min/max on both of those. If I were to do something like above, I would be duplicating that logic.